xml_converter/src/attribute/color.cpp

```diff
@@ -13,16 +14,67 @@
 
 using namespace std;
 
+////////////////////////////////////////////////////////////////////////////////
+// int_to_float // float_to_int
+//
+// Helper functions to convert the value of colors from one type to another.
+// Also serves to make sure the values stay within the bounds.
+////////////////////////////////////////////////////////////////////////////////
+float int_to_float(int input) {
+    if (input > 255) {
+        input = 255;
+    }
+    if (input < 0) {
+        input = 0;
+    }
+    float output = static_cast<float>(input) / 255.0f;
+    return output;
+}
+
+int float_to_int(float input) {
+    if (input > 1.0) {
+        input = 1.0;
+    }
+    if (input < 0) {
+        input = 0;
+    }
+    int output = static_cast<int>(input * 255);
+    return output;
+}
+
 ////////////////////////////////////////////////////////////////////////////////
 // parse_color
 //
 // Parses a Color from the value of a rapidxml::xml_attribute.
-// TODO(#98): Color should be saved in a better format then the raw hex string.
-// TODO(#129): Compound Value XML Export
 ////////////////////////////////////////////////////////////////////////////////
 Color parse_color(rapidxml::xml_attribute<>* input, vector<XMLError*>*) {
     Color color;
-    color.hex = get_attribute_value(input);
+    std::string input_string = get_attribute_value(input);
+    std::string hex_string;
+
+    if (input_string.size() > 0 && input_string[0] == '#') {
+        hex_string = input_string.substr(1);
+    }
+    else {
+        hex_string = input_string;
+    }
+
+    std::regex hex_pattern("^([A-Fa-f0-9]{6})$|^([A-Fa-f0-9]{8})$");
```

Comment:
Because you are looking for 6 or 8 hex characters later you can just look for "one or more" hex character using `+`